File: Apps/Promaker/Promaker/Knowledge/ServerEventNames.cs
namespace Promaker.Knowledge;

/// <summary>
/// **Phase S7 D-S7-2c (s6-r32) — SSE event name SSOT** (client-side).
/// <para/>
/// server-side 의 F# <c>Ds2.LightHouseService.ServerEventNames</c> module 과 wire string 정합 의무.
/// 양쪽 SSOT 가 분리된 사유 = K4 Protocol SSOT 통합 phase 가 별 phase (todo-lighthouse-kb-server.md §7.7 K4)
/// — 본 phase 까지는 양쪽 박제, K4 통합 시 `Ds2.LightHouse.Protocol` neutral project 의 단일 SSOT 로 합침.
/// <para/>
/// 신규 event 추가 시 server <c>EventBus.fs</c> 의 <c>module ServerEventNames</c> + 본 class 둘 다 1 줄씩 박제.
/// </summary>
public static class ServerEventNames
{
    public const string CollectionAdded = "collection-added";
    public const string CollectionUpdated = "collection-updated";
    public const string CollectionDeleted = "collection-deleted";
    public const string Keepalive = "keepalive";

    /// <summary>
    /// **D-S7-2c (s6-r32)** — client → server publish (POST /events/caption-progress) → server fan-out.
    /// Phase 2 vision caption indexing 진행률 박제 path.
    /// </summary>
    public const string CaptionProgress = "caption-progress";
}

File: Solutions/Core/Ds2.LightHouse.Protocol/ServerEventNames.fs
namespace Ds2.LightHouse.Protocol

/// **K4 Protocol SSOT (A2)** — SSE event name 단일 박제.
///
/// **본 module 박제 이전** = server F# `Ds2.LightHouseService.ServerEventNames` (`EventBus.fs`) +
/// client C# `Promaker.Knowledge.ServerEventNames` (`ServerEventNames.cs`) 양분. K4 통합으로
/// 단일 SSOT — 새 event 추가 시 본 module 1회 박제만으로 양측 컴파일러 강제.
///
/// 신규 event 추가 시 본 module 의 `[<Literal>]` 1줄만 박제 — server `ServerEvent` factory / client
/// subscriber 분기 caller 가 동일 SSOT 참조.
module ServerEventNames =

    [<Literal>]
    let CollectionAdded = "collection-added"

    [<Literal>]
    let CollectionUpdated = "collection-updated"

    [<Literal>]
    let CollectionDeleted = "collection-deleted"

    [<Literal>]
    let Keepalive = "keepalive"

    /// **D-S7-2c (s6-r32)** — client → server publish (Phase 2 vision caption 진행률, `POST /events/caption-progress`)
    /// → server fan-out.
    [<Literal>]
    let CaptionProgress = "caption-progress"

    /// **D-S7-5 (s6-r60)** — resumable upload PATCH 진행률. CollectionId 필드 = uploadId (wire 재사용).
    [<Literal>]
    let UploadProgress = "upload-progress"
